### PR TITLE
feat: Add ability to fetch number of sequences and I-th sequence from FAI index

### PR DESCRIPTION
**Pull request by jamorrison**
`faidx.c` has two functions for retrieving the number of sequences in the FAI index (`faidx_nseq`) and the name of the I-th sequence (`faidx_iseq`). Previously these two functions weren't accessible through `rust-htslib`. I've added a function to access each of these into the `faidx` module.

This is my first PR for `rust-htslib`, so let me know if there are any things I can improve upon in my code!

### CODE REVIEW

File: src/faidx/mod.rs [hunk 1]
```diff
@@ -99,6 +99,33 @@ impl Reader {
         let bytes = self.fetch_seq(name, begin, end)?;
         Ok(std::str::from_utf8(bytes).unwrap().to_owned())
     }
+
+    /// Fetches the number of sequences in the fai index
+    pub fn fetch_n_seqs(&self) -> u64 {
```
Comment (johanneskoester) [inline src/faidx/mod.rs:104]:
```suggestion
    pub fn n_seqs(&self) -> u64 {
```

File: src/faidx/mod.rs [hunk 2]
```diff
@@ -99,6 +99,33 @@ impl Reader {
...
+        let n = unsafe { htslib::faidx_nseq(self.inner) };
+        n as u64
+    }
+
+    /// Fetches the i-th sequence name
+    ///
+    /// # Arguments
+    ///
+    /// * `i` - index to query
+    pub fn fetch_i_seq(&self, i: i32) -> Result<String> {
```
Comment (johanneskoester) [inline src/faidx/mod.rs:114]:
```suggestion
    pub fn seq_name(&self, i: i32) -> Result<String> {
```

**Review by johanneskoester — CHANGES_REQUESTED** [review]
Thanks! I would rename the functions a bit, see below.

**Comment by brentp** [comment]
I need this functionality, anything holding up the PR?

**Comment by brentp** [comment]
Thanks @brainstorm . 
@jamorrison , you can take my tests nearly verbatim from #381

**Comment by jamorrison** [comment]
Thanks @brentp for the test code!

@johanneskoester I've added added tests for the two functions in the PR. Let me know if there's further work that needs to be done to wrap up this PR.

**Review by johanneskoester — APPROVED** [review]

**Review by johanneskoester — APPROVED** [review]

**Pull request merged**